react/src/pages/utils/components/GridFieldsDisplay.tsx

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import { Grid, GridProps } from "@material-ui/core";
+import { Field } from "../typings";
+import FieldDisplayEditable from "./FieldDisplayEditable";
+
+// Grids are very picky about what sizes are allowed
+type Width = Exclude<GridProps["xs"], boolean | "auto" | undefined>;
+
+/**
+ * Display a collection of editable fields. Number of columns can be specified.
+ *
+ * @param orderPriority How to display the fields (left-right first or top-down first)
+ * @param columnWidth The width of each column (out of 12, must be an integer)
+ */
+export default function GridFieldsDisplay(props: {
+    fields: Field[];
+    editMode: boolean;
+    onEdit: (fieldName: string | undefined, value: any) => void;
+    enums: any;
+    orderPriority?: "left-right" | "top-down";
```

Comment:
This might be better as a default value `orderPriority = "top-down"` but that means the typing for the props will need to be separated or the arguments will look really confusing. Alternatively document in the comment above what the defaults are.